### ISSUE TITLE
Sklearn type models in Nx

### ISSUE THREAD

**Issue by ivan-rivera** [elixir-nx/nx#232]
Hi all :)

There seems to be a lot of interest in building up support for neural networks, but I'm wondering if anyone is interested in collaborating on something akin to Python's [scikit-learn](https://scikit-learn.org/stable/). We could begin with a few basic algorithms like linear and logistic regressions, decision trees, random forests and gradient boosting. We could also include some unsupervised learning like k-means clustering.

While these methods might not be as exciting as NNs, they tend to be more user-friendly and they usually perform better on simple problems, so I'd think the Elixir community would appreciate having these tools available to use.

NX is still very young, so I'm not sure what would make more sense: creating a new library based on NX or creating a set of PRs to embed these algorithms within NX.

Would love to hear your thoughts!

**Comment by josevalim** [comment]
Hi @ivan-rivera! This is one area that we are sure that it should not be part of Nx - or at least not for now. However, we would love if someone could start implementing those algorithms with Nx, both to find holes in our API support, and because implementing them with Nx means they get to run on the GPU for free. So it would be great if interested folks like you start tackling this, yes!

**Comment by seanmor5** [comment]
@ivan-rivera I think a version of some of the algorithms in sklearn would be excellent, especially from a teaching perspective! Most people get introduced to ML in Python through a lot of those beginner friendly algorithms, so having a version to introduce users to Nx and Elixir would be great. I know @elbow-jason was interested in porting something similar. You might find other interested people in the [Nx mailing list](https://groups.google.com/g/elixir-nx) or the Nx channel of the Elixir slack.

**Comment by ivan-rivera** [comment]
@josevalim, @seanmor5 awesome, thanks for sharing your thoughts! I'm going to close this issue then

**Comment by elbow-jason** [comment]
@ivan-rivera I have started porting part of SciPy over to Nx with the eventual goal of getting at least the ensemble part of sklearn into Nx as well. I've got a lot going on at the moment, but I should be putting the work into a public repo in the next few days. I would love to get your help if your are interested.

**Comment by ivan-rivera** [comment]
> @ivan-rivera I have started porting part of SciPy over to Nx with the eventual goal of getting at least the ensemble part of sklearn into Nx as well. I've got a lot going on at the moment, but I should be putting the work into a public repo in the next few days. I would love to get your help if your are interested.

Definitely!

**Comment by t-rutten** [comment]
@ivan-rivera @elbow-jason I'd be very interested in helping out!
